DESIGN.md

```diff
@@ -0,0 +1,416 @@
+# OAK: The Object ArKive
+
+## All that to avoid JSON?
+
+OAK is a serialization and envelope format which encodes simple Ruby
+objects as strings.  It bundles together a variety of well-understood
+encoding libraries into a succinct self-describing package.
+
+This document covers the existing OAK format, and proposed encryption
+extension.
+
+OAK compares to JSON, YAML, and Marshal.  OAK is more precise than
+JSON or YAML, but slightly more Ruby-esque, and supports fewer types
+than Marshal.  OAK also has features similar to OpenPGP
+([https://tools.ietf.org/html/rfc4880](https://tools.ietf.org/html/rfc4880))
+(though not, so far, encryption).
+
+The main value proposition for OAK is operational flexibility.  OAK
+leds you defer choices between compression, checksumming, and 7-bit
+cleanliness algorithms until after a system is live and under load.
+
+As of 2017-09-13, OAK is used by ALI for volatile caches in Redis, and
+for durable Correspondence bodies in S3.
+
+Author:
+- [jhw@prosperworks.com](mailto:jhw@prosperworks.com)
+
+Advisors:
+- [marshall@prosperworks.com](mailto:marshall@prosperworks.com)
+- [gerald@prosperworks.com](mailto:gerald@prosperworks.com)
```

Comment:
Might as well not list Gerald here, unless he is agreeing to continue to support Oak, in which case we might want to update his contact info